app/controllers/application_controller.rb

```diff
@@ -1,3 +1,22 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  helper_method :sign_in?, :sign_in_user
+
+  def sign_in?
+    !session[:user_id].nil?
```

Comment:
Not nil means usually true. `session[:user_id]` would act in similar way, however I would not consider it 100% as signed in user. I would recommend you to fetch the user and if that result is a User object its signed in. Check out the solutions.